modules/Oculus_module/src/OculusModule.cpp

```diff
@@ -932,6 +994,43 @@ bool OculusModule::updateModule()
 
         if (buttonMapping > 0)
         {
+            if (m_useOpenXr)
+            {
+                if (!m_frameTransformInterface->frameExists(m_headFrameName))
+                {
+                    yError() << "[OculusModule::updateModule] The frame named " << m_headFrameName
+                             << " does not exist.";
+                    yError() << "[OculusModule::updateModule] I will not start the walking. Please "
+                                "try to start again.";
+                    return true;
+                }
+
+                yarp::sig::Matrix openXrHeadInitialTransform = identitySE3();
+                if (!m_frameTransformInterface->getTransform(
+                        m_headFrameName, m_rootFrameName, openXrHeadInitialTransform))
+                {
+                    yError() << "[OculusModule::updateModule] Unable to evaluate the "
+                             << m_headFrameName << " to " << m_rootFrameName << "transformation";
+                    yError() << "[OculusModule::updateModule] I will not start the walking. Please "
+                                "try to start again.";
+                    return true;
```

Comment:
@S-Dafarra  here we are returning `true` because we would like to _wait_ until `getTransform(...)` returns true or there are some other reasons?

Reply:
That ``if`` is in the ``updateModule`` method and it is reached when starting the state machine. If we have that error, we return true to avoid continuing (thus the internal state machine will not change state), and we avoid closing the module (as it happens when returning false).

Reply:
Yes, so I understood correctly your intention that it was intentionally set to true to avoid closing the module and wait for the transformation :)